### ISSUE TITLE
PR# 217 didn't fix issue with imjournal for EL 6

### ISSUE THREAD

**Issue by phamp008** [saz/puppet-rsyslog#234]
```
Oct  5 22:26:41 node1 rsyslogd: [origin software="rsyslogd" swVersion="5.8.10" x-pid="3998" x-info="http://www.rsyslog.com"] start
Oct  5 22:26:41 node1 rsyslogd-3003: invalid or yet-unknown config file command - have you forgotten to load a module? [try http://www.rsyslog.com/e/3003 ]
Oct  5 22:26:41 node1 rsyslogd: the last error occured in /etc/rsyslog.conf, line 40:"$imjournalStateFile "
Oct  5 22:26:41 node1 rsyslogd-3003: invalid or yet-unknown config file command - have you forgotten to load a module? [try http://www.rsyslog.com/e/3003 ]
Oct  5 22:26:41 node1 rsyslogd: the last error occured in /etc/rsyslog.conf, line 41:"$imjournalIgnorePreviousMessages "
Oct  5 22:26:41 node1 rsyslogd-3003: invalid or yet-unknown config file command - have you forgotten to load a module? [try http://www.rsyslog.com/e/3003 ]
Oct  5 22:26:41 node1 rsyslogd: the last error occured in /etc/rsyslog.conf, line 42:"$imjournalRatelimitInterval "
Oct  5 22:26:41 node1 rsyslogd-3003: invalid or yet-unknown config file command - have you forgotten to load a module? [try http://www.rsyslog.com/e/3003 ]
Oct  5 22:26:41 node1 rsyslogd: the last error occured in /etc/rsyslog.conf, line 43:"$imjournalRatelimitBurst "
Oct  5 22:26:41 node1 rsyslogd-2124: CONFIG ERROR: could not interpret master config file '/etc/rsyslog.conf'. [try http://www.rsyslog.com/e/2124 ]

```

```
# Settings for imjournal (If supported)
$imjournalStateFile
$imjournalIgnorePreviousMessages
$imjournalRatelimitInterval
$imjournalRatelimitBurst

```

**Comment by TheMeier** [comment]
same on debian, looks like the undef checks in the erb are wrong

from https://docs.puppet.com/puppet/latest/reference/lang_template_erb.html#testing-for-undefined-variables
Also, note that Puppet’s special undef value becomes Ruby’s special nil value in ERB templates.

**Comment by thlapin** [comment]
I've created #235 which includes a test for this  as well.

**Comment by saz** [comment]
Can you give the current master a try, if this issue is resolved now, please?

**Comment by phamp008** [comment]
Same running module 4.0.3
```
├── saz-limits (v2.3.0)
├── saz-resolv_conf (v3.0.5)
├── saz-rsyslog (v4.0.3)

```
/var/log/messages - when restarting rsyslog service
```
Nov  3 23:10:40 <server_name> kernel: Kernel logging (proc) stopped.
Nov  3 23:10:40 <server_name> rsyslogd: [origin software="rsyslogd" swVersion="5.8.10" x-pid="2272" x-info="http://www.rsyslog.com"] exiting on signal 15.
Nov  3 23:10:41 <server_name> kernel: imklog 5.8.10, log source = /proc/kmsg started.
Nov  3 23:10:41 <server_name> rsyslogd: [origin software="rsyslogd" swVersion="5.8.10" x-pid="2929" x-info="http://www.rsyslog.com"] start
Nov  3 23:10:41 <server_name> rsyslogd-3003: invalid or yet-unknown config file command - have you forgotten to load a module? [try http://www.rsyslog.com/e/3003 ]
Nov  3 23:10:41 <server_name> rsyslogd: the last error occured in /etc/rsyslog.conf, line 40:"$imjournalStateFile "
Nov  3 23:10:41 <server_name> rsyslogd-3003: invalid or yet-unknown config file command - have you forgotten to load a module? [try http://www.rsyslog.com/e/3003 ]
Nov  3 23:10:41 <server_name> rsyslogd: the last error occured in /etc/rsyslog.conf, line 41:"$imjournalIgnorePreviousMessages "
Nov  3 23:10:41 <server_name> rsyslogd-3003: invalid or yet-unknown config file command - have you forgotten to load a module? [try http://www.rsyslog.com/e/3003 ]
Nov  3 23:10:41 <server_name> rsyslogd: the last error occured in /etc/rsyslog.conf, line 42:"$imjournalRatelimitInterval "
Nov  3 23:10:41 <server_name> rsyslogd-3003: invalid or yet-unknown config file command - have you forgotten to load a module? [try http://www.rsyslog.com/e/3003 ]
Nov  3 23:10:41 <server_name> rsyslogd: the last error occured in /etc/rsyslog.conf, line 43:"$imjournalRatelimitBurst "
Nov  3 23:10:41 <server_name> rsyslogd-2124: CONFIG ERROR: could not interpret master config file '/etc/rsyslog.conf'. [try http://www.rsyslog.com/e/2124 ]

```
/etc/rsyslog.conf
```
# Settings for imjournal (If supported)
$imjournalStateFile
$imjournalIgnorePreviousMessages
$imjournalRatelimitInterval
$imjournalRatelimitBurst

```

**Comment by saz** [comment]
Can you try it with the current master? 


-saz


Sent using Boxer

On 4 Nov 2016 12:20 AM, phamp008 <notifications@github.com> wrote:

Same running module 4.0.3

├── saz-limits (v2.3.0) ├── saz-resolv_conf (v3.0.5) ├── saz-rsyslog (v4.0.3) 

/var/log/messages - when restarting rsyslog service

Nov 3 23:10:40 <server_name> kernel: Kernel logging (proc) stopped. Nov 3 23:10:40 <server_name> rsyslogd: [origin software="rsyslogd" swVersion="5.8.10" x-pid="2272" x-info="http://www.rsyslog.com"] exiting on signal 15. Nov 3 23:10:41 <server_name> kernel: imklog 5.8.10, log source = /proc/kmsg started. Nov 3 23:10:41 <server_name> rsyslogd: [origin software="rsyslogd" swVersion="5.8.10" x-pid="2929" x-info="http://www.rsyslog.com"] start Nov 3 23:10:41 <server_name> rsyslogd-3003: invalid or yet-unknown config file command - have you forgotten to load a module? [try http://www.rsyslog.com/e/3003 ] Nov 3 23:10:41 <server_name> rsyslogd: the last error occured in /etc/rsyslog.conf, line 40:"$imjournalStateFile " Nov 3 23:10:41 <server_name> rsyslogd-3003: invalid or yet-unknown config file command - have you forgotten to load a module? [try http://www.rsyslog.com/e/3003 ] Nov 3 23:10:41 <server_name> rsyslogd: the last error occured in /etc/rsyslog.conf, line 41:"$imjournalIgnorePreviousMessages " Nov 3 23:10:41 <server_name> rsyslogd-3003: invalid or yet-unknown config file command - have you forgotten to load a module? [try http://www.rsyslog.com/e/3003 ] Nov 3 23:10:41 <server_name> rsyslogd: the last error occured in /etc/rsyslog.conf, line 42:"$imjournalRatelimitInterval " Nov 3 23:10:41 <server_name> rsyslogd-3003: invalid or yet-unknown config file command - have you forgotten to load a module? [try http://www.rsyslog.com/e/3003 ] Nov 3 23:10:41 <server_name> rsyslogd: the last error occured in /etc/rsyslog.conf, line 43:"$imjournalRatelimitBurst " Nov 3 23:10:41 <server_name> rsyslogd-2124: CONFIG ERROR: could not interpret master config file '/etc/rsyslog.conf'. [try http://www.rsyslog.com/e/2124 ] 

/etc/rsyslog.conf

# Settings for imjournal (If supported) $imjournalStateFile $imjournalIgnorePreviousMessages $imjournalRatelimitInterval $imjournalRatelimitBurst

**Comment by phamp008** [comment]
I don't understand your question, our Puppet server is pulling update from latest greatest version
```
  "name": "saz-rsyslog",
  "version": "4.0.3",
  "author": "saz",
  "summary": "Manage rsyslog client and server",
  "license": "Apache-2.0",
  "source": "https://github.com/saz/puppet-rsyslog.git",
  "project_page": "https://github.com/saz/puppet-rsyslog",
  "issues_url": "https://github.com/saz/puppet-rsyslog/issues",

```

**Comment by saz** [comment]
This should be fixed in v5.0.0

**Comment by phamp008** [comment]
@saz  thank you ! I confirmed issue is resolved
